### PR TITLE
[SPIR-V][Doc] Add JointMatrixWorkItemLengthINTEL instruction to joint matrix extension

### PR DESCRIPTION
**Pull request by MrSidims**
Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

File: sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc [hunk 1]
```diff
@@ -296,6 +297,36 @@ specifying the <<Memory_Operands,*memory operand*>> *None*. +
...
+<<OpJointMatrixWorkItemLengthINTEL,*OpJointMatrixWorkItemLengthINTEL*>>,
+behavior is undefined.
+
+[cols="1,1,3*3",width="100%"]
+|=====
+4+|[[OpJointMatrixWorkItemLengthINTEL]]*OpJointMatrixWorkItemLengthINTEL* +
+ +
+Return number of components owned by the current work-item in a joint matrix. +
+ +
+'Result Type' must be an unsigned integer type. +
```
Comment (bashbaug) [inline sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc:316]: At some point we may want to clarify that the result type must be an unsigned integer-type **scalar**, unless we think there is a use-case for a vector result type.

Do we want to support arbitrary integer sizes?  I was comparing this instruction to **OpSizeOf** just to understand the differences, and for what it's worth **OpSizeOf** says:

> Result Type must be a 32-bit integer type scalar.

**Review by bashbaug — APPROVED** [review]

File: sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc (hunk 1 above)
Comment (MrSidims) [reply sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc:316]: Thanks Ben! I think, that Result should fir in one SPIR-V Word aka be 32-bit integer. I'll update the spec.

**Review by bashbaug — DISMISSED** [review]

File: sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc [hunk 2]
```diff
@@ -296,6 +297,36 @@ specifying the <<Memory_Operands,*memory operand*>> *None*. +
 'Memory Access'
 |=====
 
+==== 3.37.12. Composite Instructions
+
+Modify *OpVectorExtractDynamic* and *OpVectorInsertDynamic* to accept
+<<OpTypeJointMatrixINTEL,*OpTypeJointMatrixINTEL*>> as the 'Vector' operand.
+In this case the instructions operate on an implicit vector which repesents
```
Comment (dkhaldi) [inline sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc:304]: typo in "represents"

File: sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc [hunk 3]
```diff
@@ -296,6 +297,36 @@ specifying the <<Memory_Operands,*memory operand*>> *None*. +
 'Memory Access'
 |=====
 
+==== 3.37.12. Composite Instructions
```
Comment (dkhaldi) [inline sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc:300]: Why this section is numbered 3.37.12 instead of 3.37.9?

**Comment by bader** [comment]
@MrSidims, could you address the comments from @dkhaldi, please?

**Comment by MrSidims** [comment]
> @MrSidims, could you address the comments from @dkhaldi, please?

Thanks for pinging! Done

**Review by dkhaldi — APPROVED** [review]
LGTM

**Comment by MrSidims** [comment]
@bashbaug could you please re-approve the PR?

@bader can we merge it?

**Comment by MrSidims** [comment]
@bader should we wait for @bashbaug re-approval here?

**Comment by bader** [comment]
I don't think it's required for a simple typo fix.

**Comment by MrSidims** [comment]
@bader can we merge this PR then? :)

**Pull request merged**